consensus/src/threshold_simplex/verifier.rs

```diff
@@ -172,39 +155,29 @@ pub fn verify_finalization<
     };
     let public_key = poly::public(polynomial);
 
-    // Parse signature
-    let Some(signature) = group::Signature::deserialize(&finalization.proposal_signature) else {
+    // Parse signatures
+    let Some(finalization_signature) =
+        group::Signature::deserialize(&finalization.proposal_signature)
+    else {
         debug!(reason = "invalid signature", "dropping finalization");
         return false;
     };
-
-    // Verify threshold finalization
-    let message = proposal_message(proposal.view, proposal.parent, &payload);
-    if primitives::ops::verify_message(
-        &public_key,
-        Some(finalization_namespace),
-        &message,
-        &signature,
-    )
-    .is_err()
-    {
-        debug!(reason = "invalid signature", "dropping finalization");
-        return false;
-    }
-    debug!(view = proposal.view, "finalization verified");
-
-    // Verify seed
-    let seed = seed_message(proposal.view);
-    let Some(signature) = group::Signature::deserialize(&finalization.seed_signature) else {
+    let Some(seed_signature) = group::Signature::deserialize(&finalization.seed_signature) else {
         debug!(reason = "invalid seed signature", "dropping finalization");
         return false;
     };
-    if primitives::ops::verify_message(&public_key, Some(seed_namespace), &seed, &signature)
-        .is_err()
-    {
-        debug!(reason = "invalid seed signature", "dropping finalization");
+
+    // Verify aggregate signature
+    let signature = aggregate_signatures(&[finalization_signature, seed_signature]);
+    let finalization_message = proposal_message(proposal.view, proposal.parent, &payload);
+    let finalization_message = (Some(finalization_namespace), finalization_message.as_ref());
+    let seed_message = seed_message(proposal.view);
+    let seed_message = (Some(seed_namespace), seed_message.as_ref());
+    let messages = [finalization_message, seed_message];
+    if aggregate_verify_multiple_messages(public_key, &messages, &signature, 1).is_err() {
```

Comment:
While it is tempting to put this in `spawn_blocking`, I think that would necessitate us changing our philosophy on all signature verification not being in the async handler. That would be a big change.